### ISSUE TITLE
Implement error states in the Admin UI

### ISSUE THREAD

**Issue by JedWatson** [keystonejs/keystone#3041]
We need to improve our handling of various different error states in the Admin UI.

The ones I can think of are:

## Common

- [ ] Server error - if we get an error (that's not in the list of expected cases below) from the API, it should display an error message with a button to resend the request (not poll every few seconds like we currently do)

## List View

- [ ] Invalid List - should display that the `listKey` (from the URL) doesn't match any registered lists, and have a big button to go back to the home screen.
- [ ] Invalid Columns - if you've specified columns that can't be expanded, it should display a warning and a button that resets the columns to default
- [ ] Invalid Filters - if you've specified filters that can't be expanded, it should display a warning and a button that resets the filters

## Items View

- [ ] Invalid List - should display that the `listKey` (from the URL) doesn't match any registered lists, and have a big button to go back to the home screen.
- [ ] Invalid Item ID (404) - should display the 404 and have a link to go back to the List view. We could improve on that by having a search field as well, that submits back to the List view
- [ ] Invalid Relationships Config - I'm not sure where this is at, we had a message that would be displayed below the Edit Form (back when we were rendering Jade templates) that displayed configuration issues with Relationships that prevented them from being displayed. We should look at re-implementing that if it's been lost, as relationships are way too easy to misconfigure.

**Comment by mxstbr** [comment]
Agreed, currently all error states are the same which isn't perfect.

**Comment by JedWatson** [comment]
Ref #2846 and #3013

**Comment by mxstbr** [comment]
For invalid list, we throw a reference error in [lib/core/list.js:7](https://github.com/keystonejs/keystone/blob/c9a244dd2af4599186042a77fd26f621c9012b7a/lib/core/list.js#L7) when a list doesn't exist, which gets returned from the server as a 500.

Should we change that behavior to render the frontend and show a custom error?
